code/src/java/pcgen/gui3/dialog/NewPurchaseMethodDialogController.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2019 (C) Eitan Adler <lists@eitanadler.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.     See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+ */
+
+package pcgen.gui3.dialog;
+
+import pcgen.cdom.base.Constants;
+
+import javafx.application.Platform;
+import javafx.event.ActionEvent;
+import javafx.fxml.FXML;
+import javafx.scene.Scene;
+import javafx.scene.control.Alert;
+import javafx.scene.control.Label;
+import javafx.scene.control.Slider;
+import javafx.scene.control.TextField;
+import javafx.stage.Window;
+
+// consider using Dialog rather than implementing this ourselves
+public class NewPurchaseMethodDialogController
+{
+	private final NewPurchaseMethodModel model = new NewPurchaseMethodModel();
+	@FXML
+	private Scene newPurchaseDialog;
+	@FXML
+	private TextField nameEdit;
+
+	// should this be a slider or other numeric thing?
+	@FXML
+	private Slider pointsEdit;
+	@FXML
+	private Label shownCount;
+
+	@FXML
+	void initialize()
+	{
+		model.nameProperty().bind(nameEdit.textProperty());
+		model.pointsProperty().bind(pointsEdit.valueProperty());
+//		model.pointsProperty().addListener((observable, oldValue, newValue) ->
```

Comment:
Commented out in new class?

Reply:
replaced with a direct binding in FXML. I'm not entirely sure which way I like better, but it works.